src/actions/receive-profile.js

```diff
@@ -281,7 +282,10 @@ export function finalizeFullProfileView(
     const hasUrlInfo = maybeSelectedThreadIndexes !== null;
 
     const globalTracks = computeGlobalTracks(profile);
-    const localTracksByPid = computeLocalTracksByPid(profile, getState());
+    const localTracksByPid = computeLocalTracksByPid(
+      profile,
+      getMarkerSchemaByName(getState())
```

Comment:
This was the reason of the circular dependency, right?
I was looking at it commit-by-commit and was just leaving a comment to call this `getMarkerSchemaByName` here instead of the profile-logic directory :) Then I remembered that you probably did it and looked at the complete code and indeed you did :)

Aside from the obvious benefit of not having the circular dependency, I think it's also better this way to call this selector in actions directory instead of the profile-logic directory. We try to separate the redux state related code and the logic related code. That's why we try not to handle the raw state in the profile logic as much as possible. So it's also cleaner this way :)

Reply:
No worries, it wasn't a problem. Was happy that it wasn't an issue in the end :)